### PR TITLE
flow: fix ebpf protocol hash

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

File: probe/ebpf/flow.c [hunk 1]
```diff
@@ -199,32 +199,34 @@ static inline void fill_ipv6(struct __sk_buff *skb, int offset, __u8 *dst, __u64
 	update_hash_word(hash, w);
```
Comment (nplanel) [inline probe/ebpf/flow.c]: edit the line 111  : 	layer->_hash = FNV_BASIS ^ hash_src ^ hash_dst **^ protocol**

File: probe/ebpf/flow.c [hunk 2]
```diff
@@ -199,32 +199,34 @@ static inline void fill_ipv6(struct __sk_buff *skb, int offset, __u8 *dst, __u64
...
-static inline void fill_network(struct __sk_buff *skb, __u16 protocol, int offset,
+static inline void fill_network(struct __sk_buff *skb, __u16 netproto, int offset,
 	struct flow *flow)
 {
 	struct network_layer *layer = &flow->network_layer;
 
 	int len = skb->len - sizeof(struct ethhdr);
 	int frag = 0;
 
+	char transproto = 0;
```
Comment (nplanel) [inline probe/ebpf/flow.c]: __u8

**Comment by safchain** [comment]
run skydive-functional-hw-tests
run skydive-functional-tests-backend-orientdb
run skydive-k8s-tests

**Comment by safchain** [comment]
run skydive-functional-hw-tests

**Comment by safchain** [comment]
run skydive-functional-hw-tests

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-packaging-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-packaging-tests

File: contrib/docker/Dockerfile.compile [hunk 3]
```diff
@@ -37,7 +37,7 @@ RUN chown -R $UID /root/
 WORKDIR /root/go/src/github.com/skydive-project/skydive
 
 CMD chown -R $UID /root/go/.cache/govendor \
-    && chown -R $UID /root/.cache/go-build \
+    && chown -R $UID /root/.cache/go-build
```
Comment (lebauce) [inline contrib/docker/Dockerfile.compile]: Is it intentional ?

**Comment by lebauce** [comment]
run skydive-create-docker-images

**Comment by safchain** [comment]
run skydive-create-docker-images

**Comment by safchain** [comment]
run skydive-packaging-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Pull request merged**